File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int recurse(long long number, int cnt) {
    if (number == 1) return cnt;
    if (cnt >= 500) return -1;
    
    if (number % 2 == 0) return recurse(number/2, cnt+1);
    else return recurse(number*3+1, cnt+1);
}

int solution(int num) {
    int answer = recurse(num, 0);
    return answer;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    var answer = 0;
    
    while (num != 1 && answer < 500) {
        if (num%2==0) {
            num/=2;
        }
        else if (num%2==1) {
            num*=3
            num+=1;
        }
        answer += 1;
    }
    
    if (answer >= 500) answer = -1;
    
    return answer;
}